src/app/components/Feedback/Feedback.jsx

```diff
@@ -1,62 +1,107 @@
+/* global alert */
 import React from 'react';
 import PropTypes from 'prop-types';
 import FocusTrap from 'focus-trap-react';
+import axios from 'axios';
 
 import { trackDiscovery } from '../../utils/utils';
+import appConfig from '../../data/appConfig';
+
+const initialFields = {
+  Email: '',
+  Feedback: '',
+};
 
 class Feedback extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = { showForm: false };
+    this.currentURL = (this.props.location.pathname +
+    this.props.location.hash +
+    this.props.location.search);
+
+    this.state = {
+      showForm: false,
+      fields: Object.assign({
+        URL: this.currentURL,
+      }, initialFields),
+    };
+
     this.commentText = React.createRef();
 
     this.onSubmitForm = this.onSubmitForm.bind(this);
-    this.openForm = this.openForm.bind(this);
-    this.closeForm = this.closeForm.bind(this);
+    this.toggleForm = this.toggleForm.bind(this);
     this.deactivateForm = this.deactivateForm.bind(this);
-
   }
 
   onSubmitForm(e) {
-    if (!this.commentText.value) {
+    e.preventDefault();
+    const { fields } = this.state;
+    if (!fields.Feedback && !fields.Feedback.length) {
       this.commentText.current.focus();
     } else {
-      this.setState({ showForm: false });
+      this.postForm();
+      this.setState({
+        showForm: false,
+        fields: Object.assign({ URL: this.currentURL }, initialFields),
+      });
       trackDiscovery('Feedback', 'Submit');
-      alert('Thank you, your feedback has been submitted.');
     }
   }
 
-  openForm() {
-    trackDiscovery('Feedback', 'Open');
-    this.setState({ showForm: true });
+  postForm() {
+    const { fields } = this.state;
+    axios({
+      method: 'POST',
+      url: appConfig.feedbackFormUrl,
+      data: {
+        fields,
```

Comment:
Since you still need URL here, you can add it directly:
```
data: {
   ...fields,
   URL: this.currentURL,
},
```

rather than keeping track of the URL in state since it doesn't change.

Reply:
For Airtable, we need to pass the data structured like `{fields: { Feedback: 'this is my feedback', Email: 'patron@internet.com', URL: '/scc/page_user_is_on', } }`. So I was aiming for the elegance of just passing `{ fields }`. Refactored to completely remove handling of location/url from Feedback, because it was unreliable anyway.

Reply:
Yeah, the update is better.